src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -74,13 +117,388 @@ public class ConsensusManager
                 utxoHash = await this.consensusRules.RewindAsync().ConfigureAwait(false);
             }
 
-            this.concurrentChain.SetTip(this.Tip);
+            lock (this.treeLock)
```

Comment:
We discussed this, lock is not needed here.

Reply:
yeah but no harm either, its better practice to actually lock every entry to the dictionary.
I could remove and put a disable warring tag, but prefer not to.

Reply:
Why it would be better practice? And what warning are you talking about? Who gives you a warning?